students/YingGuo/lession07/html_render.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element():
+    tag_name = "html"
+    indent = "    "
+
+    def __init__(self, content=None, **kwargs):
+        if content is None:
+            self.content = []
+        if content is not None:
+            self.content = [content]
+        self.kwargs = kwargs 
+
+
+    def append(self, new_content):
+        self.content.append(new_content)
+
+    def write_html(self):
+        with open("output_html","w") as f:
+             f.write(self.content)
```

Comment:
Minor: Do you ever use this method (`write_html`) anywhere? I don't think this method was required in the instructions or tutorial...? No big deal, and I might be mistaken again... Just wondering...